pkg/antctl/raw/multicluster/join.go

```diff
@@ -0,0 +1,280 @@
+	"antrea.io/antrea/multicluster/controllers/multicluster/commonarea"
+	"antrea.io/antrea/pkg/antctl/raw/multicluster/common"
+	multiclusterscheme "antrea.io/antrea/pkg/antctl/raw/multicluster/scheme"
+)
+
+type clusterOptions struct {
+	Namespace string `yaml:"namespace,omitempty"`
+	ClusterID string `yaml:"clusterID,omitempty"`
+	Secret    string `yaml:"secret,omitempty"`
+	APIServer string `yaml:"apiServer,omitempty"`
+}
+
+type JoinOptions struct {
+	LeaderClusterOpt clusterOptions `yaml:"leaderCluster"`
+	MemberClusterOpt clusterOptions `yaml:"memberCluster"`
+	ClusterSetID     string         `yaml:"clusterSetID"`
+	ConfigFile       string
+}
+
+type meta struct {
+	Kind       string `yaml:"kind"`
+	APIVersion string `yaml:"apiVersion"`
+}
+
+var joinOpts *JoinOptions
+
+func (o *JoinOptions) validateAndComplete() error {
+	if o.ConfigFile != "" {
+		raw, err := ioutil.ReadFile(o.ConfigFile)
+		if err != nil {
+			return err
+		}
+		typeMeta := &meta{}
+		if err := yamlUnmarshall(raw, typeMeta); err != nil {
+			return err
+		}
+		if typeMeta.Kind != "ClusterSetConfig" || typeMeta.APIVersion != "multicluster.antrea.io/v1alpha1" {
+			return fmt.Errorf("unknown apiVersion or kind in config: %s", o.ConfigFile)
+		}
+		if err := yamlUnmarshall(raw, &joinOpts); err != nil {
+			return err
+		}
+	}
+
+	if o.LeaderClusterOpt.ClusterID == "" {
+		return fmt.Errorf("the clusterid cannot be empty")
+	}
+	if o.LeaderClusterOpt.APIServer == "" {
+		return fmt.Errorf("the API server cannot be empty")
+	}
+	if o.LeaderClusterOpt.Secret == "" {
+		return fmt.Errorf("the leader Secret cannot be empty")
+	}
+	if o.LeaderClusterOpt.Namespace == "" {
+		return fmt.Errorf("the leader Namespace cannot be empty")
+	}
+	if o.ClusterSetID == "" {
+		return fmt.Errorf("the ClusterSet cannot be empty")
+	}
+	if o.MemberClusterOpt.ClusterID == "" {
+		return fmt.Errorf("the member clusterid  cannot be empty")
+	}
+	if o.MemberClusterOpt.Namespace == "" {
+		return fmt.Errorf("the member Namespace cannot be empty")
+	}
+
+	return nil
+}
+
+func yamlUnmarshall(raw []byte, v interface{}) error {
+	d := yaml.NewDecoder(bytes.NewReader(raw))
+	d.KnownFields(false)
+	return d.Decode(v)
+}
+
+var joinExamples = strings.Trim(`
+# Join the ClusterSet from a member cluster with command line options.
+  $ antctl mc join --leader-namespace=antrea-multicluster \
+				   --leader-clusterid=test-cluster-north \
+				   --secret=member-access-token \
+				   --leader-server=https://172.18.0.3:6443 \
+				   --clusterset=test-clusterset \
+				   --namespace=kube-system \
+				   --clusterid=test-cluster-east
+
+# Join the ClusterSet from a member cluster with a config file. If both command line options and config file are specified, 
+the arguments in the config file will be used.
+  $ antctl mc join --config-file ./config.yaml
+
+# Config file example:
+apiVersion: multicluster.antrea.io/v1alpha1
+kind: ClusterSetConfig
+memberCluster:
+  namespace: kube-system
```

Comment:
I think we can also leave memberCluster's Namespace as `kube-system` considering the default manifest is using it.